### PR TITLE
Added support for Lock for E3 snapshots

### PR DESCRIPTION
**Pull request by Giulio2002**

### CODE REVIEW

**Review by awskii — APPROVED** [review]

File: eth/backend.go [hunk 1]
```diff
@@ -884,6 +884,12 @@ func New(ctx context.Context, stack *node.Node, config *ethconfig.Config, logger
 			})
 		}
 
+		for _, p := range snaptype.SeedableV3Extensions() {
```
Comment (AskAlexSharov) [inline eth/backend.go:887]: we prohibit E2 files by name and E3 files by extension?

Comment (Giulio2002) [reply eth/backend.go:887]: well issue with e3 is that files has different names. we can also do prefix / if you prefer

Comment (AskAlexSharov) [reply eth/backend.go:887]: @Giulio2002 please let's make less hacks. 
you added new feature inside .lock - to allow manage lock per type. or stick to 1  ideology or remove this feature and leave .lock prohibit everything.

Comment (AskAlexSharov) [reply eth/backend.go:887]: we need make downloader easier and more strait-forward

Comment (Giulio2002) [reply eth/backend.go:887]: but it does stick to the ideology, it is just that if you look at the names of the snapshots of e3, the only thing they have not in common is the suffix .ef or prefix (e.g `domain`). the body is the same `storage`, `accounts`, Thus i cannot work with that.

**Pull request merged**